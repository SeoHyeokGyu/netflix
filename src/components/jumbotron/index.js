import React from "react";

export default function Jumbotron(){
    return (
        <p>Jumba</p>
    );
}