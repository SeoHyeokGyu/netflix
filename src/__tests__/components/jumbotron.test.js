import React from 'react';
import { render } from '@testing-library/react';
import { Jumbotron } from '../../components';
import jumboData from '../../fixtures/jumbo';

describe('<Jumbotron />', () => {
  it('renders the <Jumbotron /> with populated data', () => {
    const { container, getByText, getByAltText, getByTestId } = render(
      <Jumbotron.Container>
        {jumboData.map((item) => (
          <Jumbotron key={item.id}>
            <Jumbotron.Pane>
              <Jumbotron.Title>{item.title}</Jumbotron.Title>
              <Jumbotron.SubTitle>{item.subTitle}</Jumbotron.SubTitle>
            </Jumbotron.Pane>
            <Jumbotron.Pane>
              <Jumbotron.Image src={item.image} alt={item.alt} data-testid={`${item.id}-jumbo-image`} />
            </Jumbotron.Pane>
          </Jumbotron>
        ))}
      </Jumbotron.Container>
    );

    expect(getByText('TV로 즐기세요.')).toBeTruthy();
    expect(getByAltText('Tiger King on Netflix')).toBeTruthy();
    expect(getByTestId('1-jumbo-image')).toBeTruthy();

    expect(
      getByText(
        '스마트 TV, PalyStation, Xbox, Chromecast, Apple TV, 블루레이 플레이어 등 다양한 디바이스에서 시청하세요.'
      )
    ).toBeTruthy();
    expect(container.firstChild).toMatchSnapshot();
  });
});
