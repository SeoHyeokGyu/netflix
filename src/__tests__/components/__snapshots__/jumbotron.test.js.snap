// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Jumbotron /> renders the <Jumbotron /> with populated data 1`] = `
<div
  class="jumbotron__Container-es77m8-2 ePEMRa"
>
  <div
    class="jumbotron__Item-es77m8-0 ivFimZ"
  >
    <div
      class="jumbotron__Inner-es77m8-1 kYQANU"
      direction="row"
    >
      <div
        class="jumbotron__Pane-es77m8-3 fOJBUm"
      >
        <h1
          class="jumbotron__Title-es77m8-4 kAumwr"
        >
          TV로 즐기세요.
        </h1>
        <h2
          class="jumbotron__SubTitle-es77m8-5 jhWhow"
        >
          스마트 TV, PalyStation, Xbox, Chromecast, Apple TV, 블루레이 플레이어 등 다양한 디바이스에서 시청하세요.
        </h2>
      </div>
      <div
        class="jumbotron__Pane-es77m8-3 fOJBUm"
      >
        <img
          alt="Tiger King on Netflix"
          class="jumbotron__Image-es77m8-6 komlBM"
          data-testid="1-jumbo-image"
          src="/images/misc/home-tv.jpg"
        />
      </div>
    </div>
  </div>
  <div
    class="jumbotron__Item-es77m8-0 ivFimZ"
  >
    <div
      class="jumbotron__Inner-es77m8-1 kYQANU"
      direction="row"
    >
      <div
        class="jumbotron__Pane-es77m8-3 fOJBUm"
      >
        <h1
          class="jumbotron__Title-es77m8-4 kAumwr"
        >
          즐겨 보는 콘텐츠를 저장해 오프라인으로 시청하세요.
        </h1>
        <h2
          class="jumbotron__SubTitle-es77m8-5 jhWhow"
        >
          간편하게 저장하고 빈틈없이 즐겨보세요.
        </h2>
      </div>
      <div
        class="jumbotron__Pane-es77m8-3 fOJBUm"
      >
        <img
          alt="Watch on mobile"
          class="jumbotron__Image-es77m8-6 komlBM"
          data-testid="2-jumbo-image"
          src="/images/misc/home-mobile.jpg"
        />
      </div>
    </div>
  </div>
  <div
    class="jumbotron__Item-es77m8-0 ivFimZ"
  >
    <div
      class="jumbotron__Inner-es77m8-1 kYQANU"
      direction="row"
    >
      <div
        class="jumbotron__Pane-es77m8-3 fOJBUm"
      >
        <h1
          class="jumbotron__Title-es77m8-4 kAumwr"
        >
          다양한 디바이스에서 시청하세요.
        </h1>
        <h2
          class="jumbotron__SubTitle-es77m8-5 jhWhow"
        >
          각종 영화와 TV 프로그램을 스마트폰, 태블릿, 노트북, TV에서 무제한으로 스트리밍하세요. 추가 요금이 전혀 없습니다.
        </h2>
      </div>
      <div
        class="jumbotron__Pane-es77m8-3 fOJBUm"
      >
        <img
          alt="Money Heist on Netflix"
          class="jumbotron__Image-es77m8-6 komlBM"
          data-testid="3-jumbo-image"
          src="/images/misc/home-imac.jpg"
        />
      </div>
    </div>
  </div>
  <div
    class="jumbotron__Item-es77m8-0 ivFimZ"
  >
    <div
      class="jumbotron__Inner-es77m8-1 kYQANU"
      direction="row"
    >
      <div
        class="jumbotron__Pane-es77m8-3 fOJBUm"
      >
        <h1
          class="jumbotron__Title-es77m8-4 kAumwr"
        >
          어린이 전용 프로필을 만들어 보세요.
        </h1>
        <h2
          class="jumbotron__SubTitle-es77m8-5 jhWhow"
        >
          자기만의 공간에서 좋아하는 캐릭터와 즐기는 신나는 모험. 자녀에게 이 특별한 경험을 선물하세요. 넷플릭스 회원이라면 무료입니다.
        </h2>
      </div>
      <div
        class="jumbotron__Pane-es77m8-3 fOJBUm"
      >
        <img
          alt="kids"
          class="jumbotron__Image-es77m8-6 komlBM"
          data-testid="4-jumbo-image"
          src="/images/misc/home-kids.png"
        />
      </div>
    </div>
  </div>
</div>
`;
